galaxykit/client.py

```diff
@@ -18,32 +19,71 @@ class GalaxyClient:
     which all authentication flows.
     """
 
-    headers = {}
+    headers = None
     galaxy_root = ""
     token = ""
     docker_client = None
 
-    def __init__(self, galaxy_root, user="", password="", container_engine=""):
+    def __init__(self, galaxy_root, auth=None, container_engine=None, container_registry=None):
         self.galaxy_root = galaxy_root
-        if user != "" and password != "":
+        self.headers = {}
+
+        if auth:
+            username, password = auth
             resp = requests.post(galaxy_root + "v3/auth/token/", auth=(user, password))
             try:
                 self.token = resp.json().get("token")
             except JSONDecodeError as e:
                 print(f"Failed to fetch token: {resp.text}", file=sys.stderr)
-            self.headers = {
-                "accept": "application/json",
+            self.headers.update({
+                "Accept": "application/json",
                 "Authorization": f"Token {self.token}",
-            }
+            })
 
-            if container_engine != "":
-                container_registry = (
+            if container_engine:
+                container_registry = container_registry or \
                     urlparse(self.galaxy_root).netloc.split(":") + ":5001"
-                )
 
                 self.docker_client = dockerutils.DockerClient(
                     (user, password), container_engine, container_registry
                 )
+    
+    def _http(self, method, path, *args, **kwargs):
+        url = urljoin(self.galaxy_root, path)
+        headers = kwargs.pop("headers", self.headers)
+        resp = requests.request(method, url, headers=headers, *args, **kwargs)
+        try:
+            json = resp.json()
+        except JSONDecodeError as e:
```

Comment:
DELETE requests don't always result in a json response, so we could accidentally come up against this error even if things are working as intended.

Reply:
I'm adding a `parse_json` flag that defaults to false. we can disable it if need be.

However, I have to say, I would consider any endpoint responding without a json body to be a bug. If you know of any like this, let me know.

Reply:
I'm happy with the solution and also with the decision to consider any non-responsive endpoint to be a bug. Resolving!